src/adapter/editor-context.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+/**
+ * React context to avoid explicitly passing the editor instance all
+ * through the component hierarchy via props.
+ */
+const EditorContext = React.createContext({});
+
+/**
+ * Helper that passes the editor context in via a prop.
+ *
+ * Context can only be accessed in `render()` and in lifecycle methods
+ * (`componentDidMount()` etc). If a component needs to access context
+ * elsewhere, such as in a constructor, the context needs to be passed
+ * in via a prop.
+ */
+EditorContext.toProps = Component => {
+	const componentWithContext = props => (
+		<EditorContext.Consumer>
+			{context => <Component {...props} context={context} />}
+		</EditorContext.Consumer>
+	);
+	componentWithContext.key = Component.key;
+	return componentWithContext;
+};
+
+export default EditorContext;
```

Comment:
Not sure where we stand on this from our previous conversations on other projects... not to worry here, but should we go with named exports everywhere? Thought that was the consensus 🤷‍♀️

Reply:
Previous discussion was on https://github.com/liferay/clay/pull/1453 — summarizing:

- That PR just uses a default export.
- Some articles were shared arguing in favor of named exports.
- @julien expressed a preference for named exports.
- You noted that, at least in Clay, we pretty strictly follow a one-export-per-module, making default exports simple.
- PR got merged, possibly without enough explicit agreement on what exactly we were agreeing to.

I'm open to sticking to default exports or named exports, and I think the one-export-per-module guideline is a good rule of thumb. Being consistent about it and picking "an" option is probably more valuable than picking the "right" option. It's not too late to change our minds if we think that one was a mistake.